nautobot/ipam/models.py

```diff
@@ -596,6 +614,30 @@ def save(self, *args, **kwargs):
             parent = max(supernets, key=operator.attrgetter("prefix_length"))
             self.parent = parent
 
+        # Validate that creation of this prefix does not create an invalid parent/child relationship
+        if self.parent and self.parent.type != constants.PREFIX_ALLOWED_PARENT_TYPES[self.type]:
+            err_msg = f"{self.type.title()} prefixes cannot be created in {self.parent.type} prefixes"
+            raise ValidationError({"type": err_msg})
+
+        invalid_children = Prefix.objects.filter(
+            ~models.Q(id=self.id),
+            ~models.Q(type__in=constants.PREFIX_ALLOWED_CHILD_TYPES[self.type]),
+            parent_id=self.parent_id,
+            prefix_length__gt=self.prefix_length,
+            ip_version=self.ip_version,
+            network__gte=self.network,
+            broadcast__lte=self.broadcast,
+            namespace=self.namespace,
+        )
```

Comment:
Quick question for myself, what is this query filtering on?

Reply:
This is filtering on:

- prefixes that share my parent
`parent_id=self.parent_id,`

- and will be reparented to me
```
            prefix_length__gt=self.prefix_length,
            ip_version=self.ip_version,
            network__gte=self.network,
            broadcast__lte=self.broadcast,
            namespace=self.namespace,
```

- but are not the correct type for this parent/child relationship
`~models.Q(type__in=constants.PREFIX_ALLOWED_CHILD_TYPES[self.type]),`

Reply:
Make this a comment and 💋